code/datums/wounds/_wounds.dm

```diff
@@ -0,0 +1,312 @@
+	// in case we ever directly add wounds
+	for(var/i in L.wounds)
+		var/datum/wound/preexisting_wound = i
+		if((preexisting_wound.type == type) && (preexisting_wound != old_wound))
+			qdel(src)
+			return
+
+	victim = L.owner
+	limb = L
+	LAZYADD(victim.all_wounds, src)
+	LAZYADD(limb.wounds, src)
+	limb.update_wounds()
+	if(status_effect_type)
+		linked_status_effect = victim.apply_status_effect(status_effect_type, src)
+	SEND_SIGNAL(victim, COMSIG_CARBON_GAIN_WOUND, src, limb)
+	if(!victim.alerts["wound"]) // only one alert is shared between all of the wounds
+		victim.throw_alert("wound", /obj/screen/alert/status_effect/wound)
+
+	var/demoted
+	if(old_wound)
+		demoted = (severity <= old_wound.severity)
+
+	if(severity == WOUND_SEVERITY_TRIVIAL)
+		return
+
+	if(!(silent || demoted))
+		var/msg = "<span class='danger'>[victim]'s [limb.name] [occur_text]!</span>"
+		var/vis_dist = COMBAT_MESSAGE_RANGE
+
+		if(severity != WOUND_SEVERITY_MODERATE)
+			msg = "<b>[msg]</b>"
+			vis_dist = DEFAULT_MESSAGE_RANGE
+
+		victim.visible_message(msg, "<span class='userdanger'>Your [limb.name] [occur_text]!</span>", vision_distance = vis_dist)
+		if(sound_effect)
+			playsound(L.owner, sound_effect, 60 + 20 * severity, TRUE)
+
+	if(!demoted)
+		wound_injury()
+		second_wind()
+
+/// Remove the wound from whatever it's afflicting, and cleans up whateverstatus effects it had or modifiers it had on interaction times. ignore_limb is used for detachments where we only want to forget the victim
+/datum/wound/proc/remove_wound(ignore_limb, replaced = FALSE)
+	//TODO: have better way to tell if we're getting removed without replacement (full heal) scar stuff
+	if(limb && !already_scarred && !replaced)
+		already_scarred = TRUE
+		var/datum/scar/new_scar = new
+		new_scar.generate(limb, src)
+	if(victim)
+		LAZYREMOVE(victim.all_wounds, src)
+		if(!victim.all_wounds)
+			victim.clear_alert("wound")
+		SEND_SIGNAL(victim, COMSIG_CARBON_LOSE_WOUND, src, limb)
+	if(limb && !ignore_limb)
+		LAZYREMOVE(limb.wounds, src)
+		limb.update_wounds()
+
+/// When you want to swap out one wound for another (typically a promotion or demotion in the same wound type). First we remove the old one, then we apply the new one with a reference to the old one, then we qdel the old one fully
+/datum/wound/proc/replace_wound(new_type)
+	var/datum/wound/new_wound = new new_type
+	already_scarred = TRUE
+	remove_wound(replaced=TRUE)
+	new_wound.apply_wound(limb, old_wound = src)
+	qdel(src)
+
+/// The immediate negative effects faced as a result of the wound
+/datum/wound/proc/wound_injury()
+	return
+
+/// Additional beneficial effects when the wound is gained, in case you want to give a temporary boost to allow the victim to try an escape or last stand
+/datum/wound/proc/second_wind()
+	if(HAS_TRAIT(victim, TRAIT_NOMETABOLISM))
```

Comment:
proposal: remove this check, and set the self_consuming variable of the determination chem to TRUE

let skeletons be determined, damnit

Reply:
skeletons can't be wounded anyway